File: 백준/Silver/10845. 큐/큐.js
const fs = require("fs");
// TODO: 제출 시 경로 변환 필수 ("/dev/stdin")
const input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .replaceAll("\r", "")
  .split("\n");

// 풀이
function solution(s) {
  const arr = [];
  const answer = [];
  for (let i = 1; i < s.length; i++) {
    const order = s[i].split(" ");
    switch (order[0]) {
      case "push":
        arr.push(Number(order[1]));
        break;
      case "pop":
        arr.length === 0 ? answer.push(-1) : answer.push(arr.shift());
        break;
      case "size":
        answer.push(arr.length);
        break;
      case "empty":
        arr.length === 0 ? answer.push(1) : answer.push(0);
        break;
      case "front":
        arr.length === 0 ? answer.push(-1) : answer.push(Number(arr[0]));
        break;
      case "back":
        arr.length === 0
          ? answer.push(-1)
          : answer.push(Number(arr[arr.length - 1]));
        break;
    }
  }
  return answer.join("\n");
}

console.log(solution(input));

File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline
N = int(input())
from collections import deque


d = deque()

for _ in range(N):
    prompt = input().split()
    
    if len(prompt) > 1:
        order, num = prompt
    else:
        order = prompt[0]

    if order == "push":
        d.append(num)
    
    elif order == "pop":
        print(d.popleft() if d else -1)
    
    elif order == "size":
        print(len(d))
    
    elif order == "empty":
        print(0 if d else 1)
    
    elif order == "front":
        print(d[0] if d else -1)
    
    elif order == "back":
        print(d[-1] if d else - 1)